src/Logger/Logger.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Drupal\password_enhancements\Logger;
+
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+use Drupal\Core\Logger\RfcLoggerTrait;
+use Drupal\Core\Utility\Error;
+use Exception;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Defines module specific logger.
+ */
+final class Logger implements LoggerInterface {
+
+  use RfcLoggerTrait;
+
+  /**
+   * Logger channel.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelInterface
+   */
+  protected $loggerChannel;
+
+  /**
+   * Constructs a new logger service.
+   *
+   * @param string $channel
+   *   Logger channel.
+   * @param \Drupal\Core\Logger\LoggerChannelFactoryInterface $logger_channel_factory
+   *   Logger channel factory.
+   */
+  public function __construct(string $channel, LoggerChannelFactoryInterface $logger_channel_factory) {
```

Comment:
Well, this should not accept the `$channel` as a parameter because this way I could use your service to log messages as `user`, `swagger_ui_field_formatter`, etc modules. But otherwise, the idea is good :) As I said in our call I wanted to build something similar for the Apigee Edge module long time ago.